middleware/csrf/csrf.go

```diff
@@ -180,29 +182,56 @@ func New(config ...Config) fiber.Handler {
 		// Store the token in the context
 		c.Locals(tokenKey, token)
 
+		// Store the token in the Go context
+		ctx := context.WithValue(c.Context(), tokenKey, token)
+		ctx = context.WithValue(ctx, handlerKey, handler)
+		c.SetContext(ctx)
+
 		// Continue stack
 		return c.Next()
 	}
 }
 
-// TokenFromContext returns the token found in the context
-// returns an empty string if the token does not exist
-func TokenFromContext(c fiber.Ctx) string {
-	token, ok := c.Locals(tokenKey).(string)
-	if !ok {
-		return ""
+// TokenFromContext returns the token from context.
+// If there is no token, an empty string is returned.
+// Supported context types:
+// - fiber.Ctx: Retrieves token from Locals
+// - context.Context: Retrieves token from context values
+func TokenFromContext(c any) string {
+	switch ctx := c.(type) {
+	case fiber.Ctx:
+		if token, ok := ctx.Locals(tokenKey).(string); ok {
+			return token
+		}
+	case context.Context:
+		if token, ok := ctx.Value(tokenKey).(string); ok {
+			return token
+		}
+	default:
+		log.Errorf("Unsupported context type: %T. Expected fiber.Ctx or context.Context", c)
```

Comment:
This should be a panic(), not a logger. See https://github.com/gofiber/fiber/pull/3287